tests/distributed/test_linears.py

```diff
@@ -0,0 +1,118 @@
+from functools import partial
+
+import pytest
+import torch
+from torch import nn
+from transformers import set_seed
+
+from optimum.neuron.utils.import_utils import (
+    is_neuronx_distributed_available,
+    is_torch_xla_available,
+)
+from optimum.neuron.utils.testing_utils import is_trainium_test
+
+from .. import launch_procs
+
+
+if is_neuronx_distributed_available():
+    from neuronx_distributed.parallel_layers.layers import (
+        RowParallelLinear,
+        create_local_weight,
+    )
+
+if is_torch_xla_available():
+    import torch_xla.core.xla_model as xm
+    import torch_xla.runtime as xrt
+
+
+INPUT_SIZE = 8192
+OUTPUT_SIZE = 2048
+
+
+class SampleModel(nn.Module):
+    def __init__(self, weights_dtype):
+        super().__init__()
+        self.linear1 = nn.Linear(INPUT_SIZE, OUTPUT_SIZE, bias=False, dtype=weights_dtype)
+
+    def forward(self, x):
+        return self.linear1(x)
+
+
+class ParallelSampleModel(nn.Module):
+    def __init__(self, weights_dtype):
+        super().__init__()
+        self.linear1 = RowParallelLinear(
+            INPUT_SIZE,
+            OUTPUT_SIZE,
+            bias=False,
+            dtype=weights_dtype,
+            sequence_parallel_enabled=False,
```

Comment:
Can you try that as well?

```suggestion
            dtype=weights_dtype,
            reduce_dtype=weights_dtype,
            sequence_parallel_enabled=False,
```

The default is `torch.float32`. Maybe in the non-parallel implemetation the reduce dtype is `bfloat16`.

Reply:
Here's the params in the example they provide [here](https://github.com/aws-neuron/neuronx-distributed/blob/7b69388089e29a4210c3f88a5aa355cd41da36d4/examples/training/llama/modeling_llama_nxd.py#L184). 
You are right I can play with the reduce_dtype, I would set it to the input_dtype though, it makes more sense to me: that is the type supposed to be at output.